### PR TITLE
Replace numpy with custom algorithm to speed up coverage completeness computation

### LINKED ISSUE #106: Speed up computation of coverage completeness over a list of genes
When it comes to calculate coverage completeness the responses get slow

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
-  Fix #106

**How to prepare for test**:
- [x] Make ready a long list pf genes, comma separated

### How to test:
- Compute coverage over genes, transcripts and exons of these genes for a sample using main branch and this branch

### Expected outcome:
- [x] Measure performance (time for the app to return response.

### Review:
- [ ] Code approved by
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
Test with a longer of genes. Here is the command:

```
curl -o /dev/null -s -w 'Total: %{time_total}s\n' -X 'POST' \
  'http://localhost:8000/coverage/samples/<genes><transcripts><exons>_coverage' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "build": "GRCh37",
  "completeness_thresholds": [
    10, 20,30
  ],
  "hgnc_gene_symbols": ["ALDH2", "ACAT2", "CHRNA1", "CHRNB1", "CHRND", "CHRNE", "CHRNG", "ACO2", "ACR", "ACTC1", "ACTG2", "ACTG1", "FLNC", "ACTN2", "ACTN1", "ACVR1" ,"PML", "RFC1", "STAT3", "APRT", "ACTA1", "ACTA2", "ACTB", "ADK", "AMPD1", "AMPD2", "ATP5F1B", "AK1", "AK2", "ADCY1", "ARF1", "SLC25A4", "FDXR", "AGRN", "ALB", "ADH5", "ALDOA", "AFP", "ZFHX3", "GANAB", "NAGA", "LRPAP1", "PSEN1", "SLC3A1", "ACY1", "APP", "ANG", "AGT", "AGTR1", "ACE", "SLC4A3", "ANK2", "CD19", "CD59", "SERPINA3", "SERPINC1", "SLC9A1", "ALDH7A1", "SERPINA1", "IFNAR1", "IFNGR1", "TFAP2A", "APOA2", "APOA1", "APOC3", "APOB", "APOE", "LRP1", "NR2F2", "AQP2", "RARS1", "CYP19A1", "DDC", "ASNS", "NARS1", "ATP2A1", "ATP2B1", "ATP2B2", "ATP2A2", "ATP6V1E1", "NPPA", "NPR2", "NPR3", "CALR", "SLC4A1", "SLC4A2", "CD40", "BCL3", "ADRB2", "B2M", "BLVRA", "HTR1A", "RHCE", "B4GALNT2", "CD79A2", "MS4A1", "BMP2", "BMP4", "BMP1", "BCAT2", "RPL13", "BRCA1", "DST", "COL17A1", "CAD", "CDH15", "CDH2", "CDH3", "CTNNA2", "CAMK2A", "CAST", "PPP3CA" ,"CALM1", "CALM2", "CALM3", "CALCRL", "CACNA2D1"
  ],
  "samples": [
    "demo_sample"
  ]
}'
```

## GENES coverage endpoint
- **main branch: time 7.629067s** 
- **This branch: Total: 4.390675s**

## TRANSCRIPTS coverage endpoint
- **main branch: 37.222885s**
- **this branch: 22.364944s**

## EXONS coverage endpoint
- main branch:

File: src/chanjo2/meta/handle_d4.py [hunk 1]
```diff
@@ -64,27 +63,34 @@ def get_intervals_completeness(
...
+
+        total_region_length += stop - start
 
-    for threshold in completeness_threholds:
-        nr_complete_bases: int64 = 0
-        for per_base_depth_region in per_base_depth:
-            nr_complete_bases += (per_base_depth_region > threshold).sum()
+        for coord_coverage in d4_file.enumerate_values(chrom, start, stop):
+            for index, threshold in enumerate(completeness_threholds):
+                if coord_coverage[2][0] >= threshold:
```
Comment (henrikstranneheim) [inline src/chanjo2/meta/handle_d4.py:83]: `coord_coverage[2][0]` 

This requires some detailed knowledge about D4 files. Would be nice with a `get_` function to help explain the retrieval of the threshold value

Comment (northwestwitch) [reply src/chanjo2/meta/handle_d4.py:83]: I agree, it's not intuitive. Would it be enough with a comment explaining what coord_coverage looks like?

Comment (henrikstranneheim) [reply src/chanjo2/meta/handle_d4.py:83]: It would help, nut perhaps constants to explain the [2][0] would be enough

Comment (northwestwitch) [reply src/chanjo2/meta/handle_d4.py]: I've added both!

**Pull request merged**